src/main/java/org/jabref/gui/exporter/DeleteMetadataAction.java

```diff
@@ -0,0 +1,259 @@
+        this.dialogService = dialogService;
+        this.taskExecutor = taskExecutor;
+        this.filePreferences = filePreferences;
+        this.xmpPreferences = xmpPreferences;
+        this.embeddedBibExporter = new EmbeddedBibFilePdfExporter(databaseMode, entryTypesManager, fieldWriterPreferences);
+
+        this.executable.bind(needsDatabase(stateManager));
+    }
+
+    @Override
+    public void execute() {
+        init();
+        BackgroundTask.wrap(this::deleteMetadata)
+                      .executeWith(taskExecutor);
+    }
+
+    public void init() {
+        if (stateManager.getActiveDatabase().isEmpty()) {
+            return;
+        }
+
+        database = stateManager.getActiveDatabase().get().getDatabase();
+        // Get entries and check if it makes sense to perform this operation
+        entries = stateManager.getSelectedEntries();
+
+        if (entries.isEmpty()) {
+
+            entries = database.getEntries();
+
+            if (entries.isEmpty()) {
+                dialogService.showErrorDialogAndWait(
+                        Localization.lang("Delete metadata from PDFs"),
+                        Localization.lang("This operation requires one or more entries to be selected."));
+                shouldContinue = false;
+                return;
+            } else {
+                boolean confirm = dialogService.showConfirmationDialogAndWait(
+                        Localization.lang("Delete metadata from PDFs"),
+                        Localization.lang("Delete metadata for all PDFs in current library?"));
+                if (confirm) {
+                    shouldContinue = false;
+                    return;
+                }
+            }
+        }
+
+        errors = entriesChanged = skipped = 0;
+
+        if (optionsDialog == null) {
+            optionsDialog = new OptionsDialog();
+        }
+        optionsDialog.open();
+        //lang
+        dialogService.notify(Localization.lang("Deleting metadata..."));
+    }
+
+    private void deleteMetadata() {
+        if (!shouldContinue || stateManager.getActiveDatabase().isEmpty()) {
+            return;
+        }
+        for (BibEntry entry : entries) {
+            // Make a list of all PDFs linked from this entry:
+            List<Path> files = entry.getFiles().stream()
+                                    .map(file -> file.findIn(stateManager.getActiveDatabase().get(), filePreferences))
+                                    .filter(Optional::isPresent)
+                                    .map(Optional::get)
+                                    .filter(path -> FileUtil.isPDFFile(path))
+                                    .collect(Collectors.toList());
+            if (files.isEmpty()) {
+                skipped++;
+                Platform.runLater(() -> optionsDialog.getProgressArea()
+                                                     .appendText("  " + Localization.lang("Skipped - No PDF linked") + ".\n"));
```

Comment:
Isn't it possible to collect the messages using a [StringJoiner](https://www.baeldung.com/java-string-joiner) and then output it as a whole?